### DISCUSSION TITLE
Get if a player is still in a loading screen.

### DISCUSSION THREAD

**Discussion by GoldenGamerLP** [Minestom/Minestom#1727] [Help]
How can i see if a player is still loading or if the client has loaded the world.

My current code:
```java
List<Player> waitingPlayers = players.stream()
                .filter(player -> player.getPlayerConnection().getConnectionState() != ConnectionState.PLAY)
                .toList();
```

**Comment by Tofaa2** [comment]
Does your current code work?

**Reply by GoldenGamerLP** [reply]
Nope, i thought the Playstate.PLAY means the client is ready to recieve packets and has fully loaded the world. The wiki minecraft says that.

**Comment by emortaldev** [comment]
Try `entity.isActive()` (returns true after setInstance) as the `PlayerSpawnEvent` is when the player is fully logged in

**Reply by GoldenGamerLP** [reply]
Ah okay, is there a way to see if the entity has loaded the world + can recieve messages etc. Thats important

**Reply by emortaldev** [reply]
The moment setInstance's future is finished, the player is ready to receive packets and has loaded all the chunks it needs. It'll be useful to look at Minestom internals.

AFAIK the spawn event is triggered after setInstance future and isActive is changed

**Reply by emortaldev** [reply]
Infact the player can receive packets before setInstance, so that should be good

**Reply by GoldenGamerLP** [reply]
Alright, so isActive() means he has loaded the world and is ready to move and everything?
